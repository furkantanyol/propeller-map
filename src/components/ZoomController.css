.zoom-controller-wrapper {
  position: fixed;
  z-index: 1;
  left: 1rem;
  top: 1rem;
  display: flex;
  flex-direction: column;
  color: white;
  font-size: 25px;
  border-radius: 5px;
  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.2);
  text-shadow: 0px 0px 2px rgba(0, 0, 0, 0.2);
  font-weight: bold;
}

.zoom-controller-button {
  cursor: pointer;
  height: 50px;
  width: 50px;
  background-color: rgb(215, 215, 215);

  display: flex;
  align-items: center;
  justify-content: center;
}

.zoom-controller-button:hover {
  background-color: rgb(187, 187, 187);
}

.zoom-controller-increase {
  border-radius: 5px 5px 0 0;
  border-bottom: 1px white solid;
}

.zoom-controller-decrease {
  border-radius: 0 0 5px 5px;
}
