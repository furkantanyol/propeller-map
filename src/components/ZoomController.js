import React from 'react'
import './ZoomController.css'

export default function ZoomController({ onZoomIn, onZoomOut }) {
  return (
    <div className='zoom-controller-wrapper'>
      <div
        className='zoom-controller-button zoom-controller-increase'
        onClick={onZoomIn}
      >
        +
      </div>
      <div
        className='zoom-controller-button zoom-controller-decrease'
        onClick={onZoomOut}
      >
        -
      </div>
    </div>
  )
}
