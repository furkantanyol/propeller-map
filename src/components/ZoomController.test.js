import React from 'react'
import ZoomController from './ZoomController'

import renderer from 'react-test-renderer'

it('renders correctly', () => {
  const component = renderer.create(
    <ZoomController
      onZoomOut={jest.fn(() => undefined)}
      onZoomIn={jest.fn(() => undefined)}
    />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
