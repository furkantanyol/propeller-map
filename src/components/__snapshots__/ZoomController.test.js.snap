// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="zoom-controller-wrapper"
>
  <div
    className="zoom-controller-button zoom-controller-increase"
    onClick={[MockFunction]}
  >
    +
  </div>
  <div
    className="zoom-controller-button zoom-controller-decrease"
    onClick={[MockFunction]}
  >
    -
  </div>
</div>
`;
